astropy/coordinates/representation.py

```diff
@@ -869,6 +892,41 @@ def represent_as(self, other_class, differential_class=None):
 
             return new_rep
 
+    def transform(self, matrix, *, is_rotation=None):
+        r"""Transform coordinates using a 3x3 matrix in a Cartesian basis.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) `~numpy.ndarray`
+            A 3x3 transformation matrix, such as a rotation matrix.
+        is_rotation : float or None (optional, keyword-only)
+            If None (default), tries to infer if the matrix is rotation.
+            If bool, skips check. True assumes matrix is a rotation matrix.
```

Comment:
Probably should add a note that this will only be used for `Spherical` representations. Which I must admit makes this slightly ugly, but I'm not sure how to avoid it...

Reply:
✅